module/zfs/zfs_ioctl.c

```diff
@@ -3645,6 +3646,9 @@ zfs_ioc_rollback(const char *fsname, nvlist_t *args, nvlist_t *outnvl)
 			error = error ? error : resume_err;
 		}
 		deactivate_super(zsb->z_sb);
+	} else if ((zv = zvol_suspend(fsname)) != NULL) {
+		error = dsl_dataset_rollback(fsname, NULL, outnvl);
```

Comment:
This is much nicer.  Did you consider passing `zv` as the owner here and similarly in `dmu_recv_end()`?